PWGLF/Tasks/QC/straRecoStudy.cxx

```diff
@@ -79,27 +79,129 @@ struct preProcessMCcollisions {
+    histos.get<TH2>(HIST("h2dNContributors"))->GetYaxis()->SetBinLabel(4, "Recoed 3 times, Biggest PV"); // size 3 = 3
+    histos.get<TH2>(HIST("h2dNContributors"))->GetYaxis()->SetBinLabel(5, "Recoed 3 times, Intermediate PV");
+    histos.get<TH2>(HIST("h2dNContributors"))->GetYaxis()->SetBinLabel(6, "Recoed 3 times, Smallest PV");
+    histos.get<TH2>(HIST("h2dNContributors"))->GetYaxis()->SetBinLabel(7, "Recoed 4 times, Biggest PV"); // size 4 = 6
+    histos.get<TH2>(HIST("h2dNContributors"))->GetYaxis()->SetBinLabel(8, "Recoed 4 times, 2nd Biggest PV");
+    histos.get<TH2>(HIST("h2dNContributors"))->GetYaxis()->SetBinLabel(9, "Recoed 4 times, 3rd Biggest PV");
+    histos.get<TH2>(HIST("h2dNContributors"))->GetYaxis()->SetBinLabel(10, "Recoed 4 times, Smallest PV");
+
+    histos.get<TH2>(HIST("h2dNContributorsWithXi"))->GetYaxis()->SetBinLabel(1, "Recoed 1 time, 1st PV");      // size 1 = 0
+    histos.get<TH2>(HIST("h2dNContributorsWithXi"))->GetYaxis()->SetBinLabel(2, "Recoed 2 times, Biggest PV"); // size 2 = 1
+    histos.get<TH2>(HIST("h2dNContributorsWithXi"))->GetYaxis()->SetBinLabel(3, "Recoed 2 times, Smallest PV");
+    histos.get<TH2>(HIST("h2dNContributorsWithXi"))->GetYaxis()->SetBinLabel(4, "Recoed 3 times, Biggest PV"); // size 3 = 3
+    histos.get<TH2>(HIST("h2dNContributorsWithXi"))->GetYaxis()->SetBinLabel(5, "Recoed 3 times, Intermediate PV");
+    histos.get<TH2>(HIST("h2dNContributorsWithXi"))->GetYaxis()->SetBinLabel(6, "Recoed 3 times, Smallest PV");
+    histos.get<TH2>(HIST("h2dNContributorsWithXi"))->GetYaxis()->SetBinLabel(7, "Recoed 4 times, Biggest PV"); // size 4 = 6
+    histos.get<TH2>(HIST("h2dNContributorsWithXi"))->GetYaxis()->SetBinLabel(8, "Recoed 4 times, 2nd Biggest PV");
+    histos.get<TH2>(HIST("h2dNContributorsWithXi"))->GetYaxis()->SetBinLabel(9, "Recoed 4 times, 3rd Biggest PV");
+    histos.get<TH2>(HIST("h2dNContributorsWithXi"))->GetYaxis()->SetBinLabel(10, "Recoed 4 times, Smallest PV");
   }
 
-  void process(aod::McCollisions const& mccollisions, soa::Join<aod::Collisions, aod::McCollisionLabels> const& collisions)
+  void process(aod::McCollision const& mcCollision, soa::SmallGroups<soa::Join<aod::McCollisionLabels, aod::Collisions>> const& collisions, TracksCompleteIUMC const& tracks, aod::McParticles const& mcParticles)
   {
-    for (auto& mccollision : mccollisions) {
-      // check if collision successfully reconstructed
-      int lExists = 0;
-      for (auto& collision : collisions) {
-        if (collision.has_mcCollision()) {
-          if (mccollision.globalIndex() == collision.mcCollision().globalIndex()) {
-            lExists++;
-          }
+    int lNumberOfXi = 0;
+    for (auto& mcp : mcParticles) {
+      // mimic triggering strategy precisely
+      if (TMath::Abs(mcp.eta()) < 0.8 && mcp.pdgCode() == 3312)
+        lNumberOfXi++;
+    }
+
+    std::vector<collisionStats> collisionStatAggregator;
```

Comment:
`std::vector` has a constructor with a single int argument, that creates a vector of that size filled with default-constructed items of the vector element type, no need to emplace them in a loop.

Reply:
thanks, done!